src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4406,194 @@ subroutine cat_enkf_increments(                               &
+       ! Determine which species are Tb
+       
+       call get_select_species(1, 'Tb', N_obs_param, obs_param, N_select_species_Tb, select_species_Tb )
+       
+       N_state_max = 7
+       
+       allocate( State_incr(N_state_max,N_ens)) 
+       allocate( State_lon( N_state_max      ))
+       allocate( State_lat( N_state_max      ))
+       
+       do kk=1,N_catd     
+          
+          N_state = 3    ! for now, assume only sfmc and/or sfds obs
+          
+          ! compute increments only snow-free and non-frozen tiles
+          
+          if ( (SWE_ensavg(kk) < SWE_threshold) .and. &
+               (tp1_ensavg(kk) > tp1_threshold)) then  
+             
+             ! find observations within halo around tile kk
+             
+             halo_minlon = tile_coord(kk)%com_lon - xcompact
+             halo_maxlon = tile_coord(kk)%com_lon + xcompact
+             halo_minlat = tile_coord(kk)%com_lat - ycompact
+             halo_maxlat = tile_coord(kk)%com_lat + ycompact
+             
+             ! simple approach to dateline issue (cut halo back to at most -180:180, -90:90)
+             ! - reichle, 28 May 2013
+             
+             halo_minlon = max(halo_minlon,-180.)
+             halo_maxlon = min(halo_maxlon, 180.)
+             halo_minlat = max(halo_minlat, -90.)
+             halo_maxlat = min(halo_maxlat,  90.)
+             
+             call get_ind_obs_lat_lon_box(                               &
+                  N_obs,            Observations,                        &
+                  halo_minlon, halo_maxlon, halo_minlat, halo_maxlat,    &
+                  N_select_species, select_species(1:N_select_species),  &
+                  N_selected_obs,   ind_obs )
+             
+             if (N_selected_obs>0) then
+                
+                ! Determine if Tb observations are present
+                
+                do ii = 1,N_select_species_Tb
+                   do jj = 1,N_selected_obs
+                      if (select_species_Tb(ii) == Observations(ind_obs(jj))%species) then
+                         found_Tb_obs = .true.
+                         exit
+                      end if
+                   end do
+                   if (found_Tb_obs) exit
+                end do
+                
+                if (found_Tb_obs) then
+                   if (cat_param(kk)%poros>=PEATCLSM_POROS_THRESHOLD) then
+                      N_state = 7
+                   else
+                      N_state = 6
+                   end if
+                end if
+                
+                ! assemble State_minus
+                ! (on  input, cat_progn contains cat_progn_minus)
+                
+                if     ( N_state==3 ) then 
+                   
+                   State_incr(1,:) = (cat_progn( kk,:)%srfexc/scale_srfexc)
+                   State_incr(2,:) = (cat_progn( kk,:)%rzexc /scale_rzexc)
+                   State_incr(3,:) = (cat_progn( kk,:)%catdef/scale_catdef)
```

Comment:
I wonder if we shouldn't look at the sensitivity of the ASCAT analysis to including/excluding catdef in the state vector.  My gut feeling is that catdef shouldn't be included in the state vector, based on the history of the Tb assimilation.  It's included in update_type=2 more for historical reasons than anything else.  If catdef should be excluded from the state vector, we might want to settle this in the code sooner rather than later, and definitely before we start publishing results.  Something to discuss on Monday.